projects/addon-editor/extensions/tiptap-node-view/node-view-render.ts

```diff
@@ -0,0 +1,157 @@
+import {Injector, Input, Type} from '@angular/core';
+import {
+    Editor,
+    NodeView,
+    NodeViewProps,
+    NodeViewRenderer,
+    NodeViewRendererOptions,
+    NodeViewRendererProps,
+} from '@tiptap/core';
+import type {Node as ProseMirrorNode} from 'prosemirror-model';
+import type {Decoration} from 'prosemirror-view';
+
+import {TuiComponentRenderer} from './component-render';
+
+/**
+ * You should extend this class to create custom
+ * Tiptap's {@link https://tiptap.dev/guide/node-views NodeView} from angular component.
+ * It contains compulsory properties which component will get externally while NodeView's rendering.
+ */
+export class TuiNodeViewNgComponent implements NodeViewProps {
+    @Input() editor!: NodeViewProps['editor'];
+    @Input() node!: NodeViewProps['node'];
+    @Input() decorations!: NodeViewProps['decorations'];
+    @Input() selected!: NodeViewProps['selected'];
+    @Input() extension!: NodeViewProps['extension'];
+    @Input() getPos!: NodeViewProps['getPos'];
+    @Input() updateAttributes!: NodeViewProps['updateAttributes'];
+    @Input() deleteNode!: NodeViewProps['deleteNode'];
+}
+
+interface TuiNodeViewRendererOptions extends NodeViewRendererOptions {
+    update?: ((node: ProseMirrorNode, decorations: Decoration[]) => boolean) | null;
```

Comment:
If it's already optional, do we really need `| null`?

Reply:
remove `null`